quic/s2n-quic-transport/src/recovery/manager.rs

```diff
@@ -1040,6 +1065,42 @@ impl<Config: endpoint::Config> Manager<Config> {
         //# constant.
         max(time_threshold, K_GRANULARITY)
     }
+
+    #[inline]
+    fn check_consistency(&self, path: &Path<Config>, is_handshake_confirmed: bool) {
+        if cfg!(debug_assertions) {
+            let ack_eliciting_packets_in_flight = self
+                .sent_packets
+                .iter()
+                .any(|(_, sent_info)| sent_info.ack_elicitation.is_ack_eliciting());
+
+            let mut timer_required = ack_eliciting_packets_in_flight;
+
+            //= https://www.rfc-editor.org/rfc/rfc9002#section-6.2.2.1
+            //# it is the client's responsibility to send packets to unblock the server
+            //# until it is certain that the server has finished its address validation
+            timer_required |= !path.is_peer_validated();
+
+            //= https://www.rfc-editor.org/rfc/rfc9002#section-6.2.2.1
+            //# If no additional data can be sent, the server's PTO timer MUST NOT be
+            //# armed until datagrams have been received from the client, because
+            //# packets sent on PTO count against the anti-amplification limit.
+            timer_required &= !path.at_amplification_limit();
+
+            //= https://www.rfc-editor.org/rfc/rfc9002#section-6.2.1
+            //# An endpoint MUST NOT set its PTO timer for the Application Data
+            //# packet number space until the handshake is confirmed.
+            timer_required &= !self.space.is_application_data() || is_handshake_confirmed;
+
+            // We haven't transmitted anything in this packet space yet so the
+            // PTO timer would not be armed yet
+            timer_required &= self.time_of_last_ack_eliciting_packet.is_some();
+
+            if timer_required {
+                assert!(self.armed_timer_count() > 0);
```

Comment:
nit: might be easier to debug if you use `assert_ne` since that'll print the actual value

```suggestion
                assert_ne!(self.armed_timer_count(), 0);
```